packages/web-forms/src/components/form-layout/FormQuestion.vue

```diff
@@ -25,15 +25,36 @@ const isRangeNode = (node: ControlNode) => node.nodeType === 'range';
 const isTriggerNode = (node: ControlNode) => node.nodeType === 'trigger';
 const isUploadNode = (node: ControlNode) => node.nodeType === 'upload';
 
-const submitPressed = inject<boolean>('submitPressed', false);
+const submitPressed = inject<Ref<boolean>>('submitPressed', ref(false));
+
+const wasEverFilled = ref(false);
+
+const isEmpty = computed(() => props.question.currentState.instanceValue.trim() === '');
+
+watch(
+	isEmpty,
+	(newIsEmpty) => {
+		if (!newIsEmpty) {
+			wasEverFilled.value = true;
+		}
+	},
+	{ immediate: true }
+);
+
+const isInvalidOnSubmit = computed(
+	() => submitPressed.value && props.question.validationState.violation?.valid === false
+);
+const isEmptyAndRequiredOnSubmit = computed(
```

Comment:
Ok, now I see this doesn't have anything to do with submission, sorry! So it's really `isRequiredAndCleared` or something. I had it backwards -- I thought this PR was adding the highlight on submission but now I see the problem was with the clearing case. 

I'll let you fix the naming back and then merge!